Build fails on Sphinx 9.x

Hey there, thanks for this project!  Solves exactly a problem I'm having (private vs. public documentation).

I was able to get this working on Sphinx 8.0.2 (simply what I happened to have laying around already installed).  However, it failed on my CI agent and the only notable difference is Sphinx 9.1.0.  I updated my local installation to 9.1.0 and was able to reproduce the problem exactly.

The full output is:

```
misha@spareparts$ make clean html
Removing everything under 'build'...
Running Sphinx v9.1.0
loading translations [en]... done
making output directory... done
Read in collections ...

Extension error (sphinx_collections.collections)!

Versions
========

* Platform:         linux; (Linux-6.12.58-gentoo-x86_64-AMD_Ryzen_9_7940HS_w-_Radeon_780M_Graphics-with-glibc2.42)
* Python version:   3.13.0 (CPython)
* Sphinx version:   9.1.0
* Docutils version: 0.21.2
* Jinja2 version:   3.1.4
* Pygments version: 2.19.1

Last Messages
=============

None.

Loaded Extensions
=================

None.

Traceback
=========

      File "/home/misha/.pyenv/versions/3.13.0/lib/python3.13/site-packages/sphinx/events.py", line 452, in emit
        raise ExtensionError(
        ...<3 lines>...
        ) from exc
    sphinx.errors.ExtensionError: Handler <function collect_collections at 0x7fc6c5a3d9e0> for event 'config-inited' threw an exception (exception: 'Tags' object has no attribute 'tags')


The full traceback has been saved in:
/tmp/sphinx-err-nrfe6g1i.log

To report this error to the developers, please open an issue at <https://github.com/sphinx-doc/sphinx/issues/>. Thanks!
Please also report this if it was a user error, so that a better error message can be provided next time.
make: *** [Makefile:20: html] Error 2
```

I get the same output from running `sphinx-build -b html -t private source/ build/html/` as well.

The relevant sections of my `conf.py` are as follows:

```python
extensions = [
        "sphinx_collections",
        "sphinx_immaterial",
        "sphinx.ext.todo",
        ]
 ...

collections = {
        "private": {
            "driver": "git",
            "source": "redacted-clone-url/private.git",
            "tags": ["private"],
            },
        }
```

In my case, it's easy enough to pin Sphinx to 8.x -- but figured I'd report this.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -26,8 +26,10 @@ jobs:
           - py311-sphinx82
           - py312-sphinx74
           - py312-sphinx82
+          - py312-sphinx91
           - py313-sphinx74
           - py313-sphinx82
+          - py313-sphinx91
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v5

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -82,6 +82,7 @@ legacy_tox_ini = """
 [tox]
 env_list =
     py{310,311,312,313}-sphinx{74,82}
+    py{312,313}-sphinx91
     docs
     linkcheck
 requires =
@@ -96,6 +97,7 @@ deps =
     pytest-xdist>=3.0
     sphinx74: sphinx==7.4.*
     sphinx82: sphinx==8.2.*
+    sphinx91: sphinx==9.1.*
 extras = docs
 commands =
     pytest -n auto --tb=long tests/

--- a/sphinx_collections/collections.py
+++ b/sphinx_collections/collections.py
@@ -66,7 +66,7 @@ class Collection:
         # Check if tags are set and change active to True if this is the case
         self.tags = tags
         for tag in tags:
-            if self.app.tags.tags.get(tag):
+            if tag in self.app.tags:
                 self.active = True
 
         self._prefix = f"  {self.name}: "

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,16 +1,16 @@
 """Pytest conftest module containing common test configuration and fixtures."""
 
+from pathlib import Path
 import shutil
 
 import pytest
-from sphinx.testing.path import path
 
 pytest_plugins = "sphinx.testing.fixtures"
 
 
 def copy_srcdir_to_tmpdir(srcdir, tmp):
-    srcdir = path(__file__).parent.abspath() / srcdir
-    tmproot = tmp / path(srcdir).basename()
+    srcdir = Path(__file__).parent.resolve() / srcdir
+    tmproot = Path(tmp) / Path(srcdir).name
     shutil.copytree(srcdir, tmproot)
     return tmproot
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5,7 +5,18 @@ import shutil
 
 import pytest
 
+from sphinx_collections import collections as _collections_module
+
 pytest_plugins = "sphinx.testing.fixtures"
+
+
+@pytest.fixture(autouse=True)
+def _reset_collections_registry():
+    # Why: COLLECTIONS is a module-level list mutated by config-inited handlers,
+    # so it leaks between tests sharing an xdist worker.
+    _collections_module.COLLECTIONS.clear()
+    yield
+    _collections_module.COLLECTIONS.clear()
 
 
 def copy_srcdir_to_tmpdir(srcdir, tmp):

--- a/tests/doc_test/config_tags/conf.py
+++ b/tests/doc_test/config_tags/conf.py
@@ -1,0 +1,24 @@
+extensions = ["sphinx_collections"]
+
+collections = {
+    "active_via_tag": {
+        "driver": "string",
+        "source": "Tag activated content",
+        "target": "active_via_tag/file.txt",
+        "active": False,
+        "tags": ["enable_collection"],
+        "final_clean": False,
+    },
+    "inactive_no_tag": {
+        "driver": "string",
+        "source": "Should not appear",
+        "target": "inactive_no_tag/file.txt",
+        "active": False,
+        "tags": ["never_set_tag"],
+        "final_clean": False,
+    },
+}
+
+project = "collections tag test"
+master_doc = "index"
+exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]

--- a/tests/doc_test/config_tags/index.rst
+++ b/tests/doc_test/config_tags/index.rst
@@ -1,0 +1,4 @@
+TEST COLLECTIONS TAG ACTIVATION
+===============================
+
+Tag activated collection works.

--- a/tests/test_config_tags.py
+++ b/tests/test_config_tags.py
@@ -1,0 +1,35 @@
+from pathlib import Path
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    "test_app",
+    [{"buildername": "html", "srcdir": "doc_test/config_tags", "tags": ["enable_collection"]}],
+    indirect=True,
+)
+def test_tag_activates_collection(test_app):
+    app = test_app
+    app.build()
+
+    activated = Path(app.confdir, "_collections", "active_via_tag", "file.txt")
+    assert activated.exists()
+    assert activated.read_text() == "Tag activated content"
+
+    inactive = Path(app.confdir, "_collections", "inactive_no_tag", "file.txt")
+    assert not inactive.exists()
+
+
+@pytest.mark.parametrize(
+    "test_app",
+    [{"buildername": "html", "srcdir": "doc_test/config_tags"}],
+    indirect=True,
+)
+def test_no_tag_keeps_collection_inactive(test_app):
+    app = test_app
+    app.build()
+
+    activated = Path(app.confdir, "_collections", "active_via_tag", "file.txt")
+    inactive = Path(app.confdir, "_collections", "inactive_no_tag", "file.txt")
+    assert not activated.exists()
+    assert not inactive.exists()

